rest-api/etl/etl.sql

```diff
@@ -0,0 +1,2062 @@
+                'StreetAddress_PIICity',
+                'StreetAddress_PIIZIP')
+        AND s1.value_string IS NOT NULL
+        )
+
+        OR
+
+        (s1.question_ppi_code = 'StreetAddress_PIIState'
+        AND s1.topic_value = 'States'
+        AND s1.value_code_id IS NOT NULL
+        )
+    GROUP BY
+        s1.participant_id
+    ) f2
+    ON f1.participant_id = f2.participant_id
+;
+
+ALTER TABLE cdm.src_participant ADD KEY (latest_date_of_location);
+-- -------------------------------------------------------------------
+-- table: src_mapped
+-- -------------------------------------------------------------------
+
+DROP TABLE IF EXISTS cdm.src_mapped;
+CREATE TABLE cdm.src_mapped
+(
+    participant_id              bigint,
+    date_of_survey              datetime,
+    question_ppi_code           varchar(200),
+    question_code_id            bigint,
+    question_source_concept_id  bigint,
+    question_concept_id         bigint,
+    value_ppi_code              varchar(200),
+    topic_value                 varchar(200),
+    value_code_id               bigint,
+    value_source_concept_id     bigint,
+    value_concept_id            bigint,
+    value_number                decimal(20,6),
+    value_boolean               tinyint,
+    value_boolean_concept_id    bigint,
+    value_date                  datetime,
+    value_string                varchar(1024),
+    questionnaire_response_id   bigint,
+    unit_id                     varchar(50)
+);
+
+INSERT INTO cdm.src_mapped
+SELECT
+    src_c.participant_id                AS participant_id,
+    src_c.date_of_survey                AS date_of_survey,
+    src_c.question_ppi_code             AS question_ppi_code,
+    src_c.question_code_id              AS question_code_id,
+    COALESCE(vc1.concept_id, 0)         AS question_source_concept_id,
+    COALESCE(vc2.concept_id, 0)         AS question_concept_id,
+    src_c.value_ppi_code                AS value_ppi_code,
+    src_c.topic_value                   AS topic_value,
+    src_c.value_code_id                 AS value_code_id, 
+    COALESCE(vc3.concept_id, 0)         AS value_source_concept_id,
+    COALESCE(vc4.concept_id, 0)         AS value_concept_id,
+    src_c.value_number                  AS value_number,
+    src_c.value_boolean                 AS value_boolean,   
+    CASE
+        WHEN src_c.value_boolean = 1 THEN 45877994
+        WHEN src_c.value_boolean = 0 THEN 45878245
+        ELSE 0
+    END                                 AS value_boolean_concept_id,
+    src_c.value_date                    AS value_date,
+    src_c.value_string                  AS value_string,
+    src_c.questionnaire_response_id     AS questionnaire_response_id,
+    src_c.unit_id                       AS unit_id 
+FROM cdm.src_clean src_c
+JOIN cdm.src_participant src_p
+    ON  src_c.participant_id = src_p.participant_id
+LEFT JOIN voc.concept vc1 
+    ON  LEFT(src_c.question_ppi_code, 50) = vc1.concept_code
```

Comment:
You technically won't need the LEFT() here anymore once using short_value above. Though it doesn't hurt.

Reply:
You can remove the LEFT now.